chall02/eligero.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+
+#* ************************************************************************** */
+#*                                                                            */
+#*                                                        :::      ::::::::   */
+#*   eligero.py                                         :+:      :+:    :+:   */
+#*                                                    +:+ +:+         +:+     */
+#*   By: eligero <eligero@student.42.fr>            +#+  +:+       +#+        */
+#*                                                +#+#+#+#+#+   +#+           */
+#*   Created: 2020/04/06 14:01:31 by eligero           #+#    #+#             */
+#*   Updated: 2020/04/06 15:21:49 by eligero          ###   ########.fr       */
+#*                                                                            */
+#* ************************************************************************** */
+
+#
+# Constelación Serpens (Chall02) / String to Morse
+#
+# Solved with key-value pair dictionary as local variable in a function, due
+# to local variables are faster than global variables in python.
+# The achieved solution has fitted challenge's problem conception, by
+# blacklisting wrong inputs by using regular expressions and, or, wrong
+# invocations.
+#
+
+import sys
+import re
+
+def encrypt (message):
+	morse_dict = { 'a':'.-', 'b':'-...', 'c':'-.-.', 'd':'-..', 'e':'.',
+		'f':'..-.', 'g':'--.', 'h':'....', 'i':'..', 'j':'.---', 'k':'-.-',
+		'l':'.-..', 'm':'--', 'n':'-.', 'o':'---', 'p':'.--.', 'q':'--.-',
+		'r':'.-.', 's':'...', 't':'-', 'u':'..-', 'v':'...-', 'w':'.--',
+		'x':'-..-', 'y':'-.--', 'z':'--..'}
+	cipher = ''
+	for char in message:
+		if char >= 'a' and char <= 'z':
+			cipher += morse_dict[char]
+		elif  char >= 'A' and char <= 'Z':
+			cipher += morse_dict[char.lower()]
+		elif char == ' ':
+			cipher += ' '
+	return (cipher)
```

Comment:
You could have made this more pythonic with this oneliner
```suggestion
	return "".join(morse_dict.get(c, " ") for c in message)
```

Reply:
That's cool. I know, but I priorized readable code. But I take note of this for future challenges.

Reply:
I find my code easier to read. That said, it is my opinion, but your multiple conditions aren't easy to understand.  
A more readable solution would have been to put my one_liner in a function that has a clear name about what it does, and call it in the return close.

Reply:
Those are opinions. Both are respectable.